src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -360,6 +362,10 @@ void BaseCouplingScheme::secondExchange()
         _isTimeWindowComplete = true;
         _timeWindowStartTime += _computedTimeWindowPart;
         _computedTimeWindowPart = 0.0; // reset window
+        _totalDriftCorrection += _driftCorrection;
+        if (abs(_totalDriftCorrection) > math::NUMERICAL_ZERO_DIFFERENCE) {
+          PRECICE_WARN("preCICE has corrected the time by a total of {}. This is often necessary, if you are using very many substeps per time window over multiple time windows.", _totalDriftCorrection);
+        }
```

Comment:
Isn't this pretty much inevitable when dealing with many time steps?

Reply:
Sure. But raising a warning still makes sense, because we should tell our users that they are entering slightly unsafe ground. I think keeping track of the `_totalDriftCorrection` and telling the user if it becomes significant and also how large it is also makes sense.

Reply:
Wouldn't this duplicate though in case of multiple coupling schemes.

Also, the correction happens in a single coupling scheme, so compositional schemes with different window sizes will drift independently.

What is the expected behaviour here?

Reply:
> Wouldn't this message duplicate though in case of multiple coupling schemes. We could instead always mention the drift in the coupling scheme info log.

I would not show the drift per default. Currently, it is an extreme exception that most of our users will not have to worry about. This might change in the future, but this is hard to estimate currently.

However, I think we should mention the drift prominently, if it occurs. This will also lead to users coming back to us and we will learn more about the specific situations where we have to be careful.

I guess we should do some testing with this warning and our tutorial cases. This warning should not show up for our tutorial cases and other cases without subcycling, but I did not test this so I cannot promise. For in-depth testing I currently do not have time, but it might also be sufficient to replace the warning with an error and look for failing unit and integration tests.

> More importantly, the correction happens in a single coupling scheme, so compositional schemes with different window sizes will drift independently.
> 
> What is the expected behaviour here?

Since every coupling scheme has its own way of tracking time I expect that different drifts might occur. Again, this is a very exceptional case. I think we cannot really factor this out as long as every coupling scheme allows its own time window size.